### PR TITLE
feat(cb2-5880): sorting tests chronologically

### PR DESCRIPTION
**Pull request by leifkemp-bjss**
## Contingency Testing - Sort Tests Chronologically

Change to the technical record view to sort tests chronologically
[CB2-5880](https://dvsa.atlassian.net/browse/CB2-5880)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

File: src/app/features/tech-record/components/test-record-summary/test-record-summary.component.ts [hunk 1]
```diff
@@ -30,15 +40,38 @@ export class TestRecordSummaryComponent {
...
+          testTypeName: t.testTypeName,
+          testNumber: t.testNumber,
+          testResult: t.testResult,
+          testResultId: record.testResultId
+        };
+        arr.push(field);
+      })
+    );
+
+    return arr.sort((a, b) => new Date(b.testTypeStartTimestamp).getTime() - new Date(a.testTypeStartTimestamp).getTime());
```
Comment (tahayazici) [inline src/app/features/tech-record/components/test-record-summary/test-record-summary.component.ts:59]: As you're just converting one type of array into a different type of array, using `map()` might be more relevant to what you want to achieve.

```suggestion
    const byDate = (a: TestField, b: TestField) => new Date(b.testTypeStartTimestamp).getTime() - new Date(a.testTypeStartTimestamp).getTime();

    return this.testRecords
      .map(record => record.testTypes
        .map(testType => ({
            testTypeStartTimestamp: testType.testTypeStartTimestamp,
            testTypeName: testType.testTypeName,
            testNumber: testType.testNumber,
            testResult: testType.testResult,
            testResultId: record.testResultId
          })
        )
      )
      .flat()
      .sort(byDate);
```

**Review by tahayazici — DISMISSED** [review]
Good stuff. Left one optional suggestion.

File: src/app/features/tech-record/components/test-record-summary/test-record-summary.component.ts (hunk 1 above)
Comment (gjulien-bjss) [reply src/app/features/tech-record/components/test-record-summary/test-record-summary.component.ts]: `flatMap` is even better 🚀

**Review by gjulien-bjss — APPROVED** [review]

File: src/app/features/tech-record/components/test-record-summary/test-record-summary.component.ts (hunk 1 above)
Comment (tahayazici) [reply src/app/features/tech-record/components/test-record-summary/test-record-summary.component.ts]: (modified my initial suggestion to use `flatMap()` instead)

**Pull request merged**